### ISSUE TITLE
Check usages of empty features

### ISSUE THREAD

**Issue by ShadowJonathan** [rust-lang/rust-clippy#8402]
### What it does

Check for all "empty" features in `Cargo.toml` (`a_feature = []`), with no optional dependency name collisions, and try to find if any config attribute in the file has that feature, if it doesn't, raise.

### Lint Name

unused_crate_feature

### Category

pedantic

### Advantage

- Allow libraries to clean up unused features, which could then signal downstream that something has changed.
- Cleaner code and crates, with less "cruft"

### Drawbacks

Possible false-positives if cargo were to infer more from features in the future (other than optional dependencies and the likes)

### Example

```toml
[features]
a_thing = []
```

This feature is nowhere to be found in any source code, so this raises the warning.

**Comment by ShadowJonathan** [comment]
https://github.com/rust-lang/rust-clippy/issues/8301 and https://github.com/rust-lang/rust-clippy/issues/1614 are related to this lint
